markets/bfp-market/contracts/storage/PerpMarket.sol

```diff
@@ -159,6 +159,28 @@ library PerpMarket {
         }
     }
 
+    function minimumCredit(
+        PerpMarket.Data storage self,
+        PerpMarketConfiguration.Data storage marketConfig,
+        uint256 price
+    ) internal view returns (uint256) {
+        return self.size.mulDecimal(price).mulDecimal(marketConfig.minCreditPercent);
+    }
+
+    /// @dev Returns the markets delegated collateral value in USD.
+    function getDelegatedCollateralValueUsd(
```

Comment:
This function is quite expensive expensive... Assuming 2 delegated collaterals with 3 supported margin collateral that's 5 calls to oracles to fetch prices on every commit/settle in addition to fetching ETH/USD for keep fees, the market/USD CL price.